dask/dataframe/core.py

```diff
@@ -7282,6 +7287,10 @@ def apply_and_enforce(*args, **kwargs):
     func = kwargs.pop("_func")
     meta = kwargs.pop("_meta")
     df = func(*args, **kwargs)
+
+    if is_scalar(df) and is_series_like(meta):
+        df = type(meta)([df])
```

Comment:
I ended up removing the explicit `dtype` since it caused issues and was quite unnecessary considering that `df` is already the final result data. The casting/coercing logic of the `Series` takes care of compat stuff with nulls, etc.